### PR TITLE
8318486: Rename JavaLangAccess.xxNoRepl to xxReportError

### PR DESCRIPTION
**Pull request by liach**
Please review a patch that renames `JavaLangAccess::xxNoRepl` to `xxReportError` to explicitly indicate these APIs report encoding errors.

The old "NoRepl" suffix presumably means "No Replacement", but it has been misunderstood as "No Replication" (#16209) and misused as an array sharing API in performance optimization patches.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318486](https://bugs.openjdk.org/browse/JDK-8318486): Rename JavaLangAccess.xxNoRepl to xxReportError (**Enhancement** - P5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16260/head:pull/16260` \
`$ git checkout pull/16260`

Update a local copy of the PR: \
`$ git checkout pull/16260` \
`$ git pull https://git.openjdk.org/jdk.git pull/16260/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16260`

View PR using the GUI difftool: \
`$ git pr show -t 16260`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16260.diff">https://git.openjdk.org/jdk/pull/16260.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16260#issuecomment-1770197370)

### CODE REVIEW

**Comment by liach** [comment]
> I assume it's "no replace", meaning it reports the error rather than use the coder's replacement value.

This revelation is too important; I might have to reevaluate my choices, for many contributors have assumed that `NoRepl` was intended to avoid duplication instead of error substitution. That said, I will look for better solutions to en/decode strings with error reporting, or creating strings or accessing the arrays with trust (as seen in a recent FFM API string writing JLA addition)

**Comment by liach** [comment]
I have decided to rename `NoRepl` to `ReportError`, as it corresponds to `CodingErrorAction.REPORT`. I have replaced all `NoRepl` occurrences I've found with `ReportError`, including in tests.

File: src/java.base/share/classes/java/lang/System.java [hunk 1]
```diff
@@ -2473,22 +2473,22 @@ public Stream<ModuleLayer> layers(ClassLoader loader) {
             public int countPositives(byte[] bytes, int offset, int length) {
                 return StringCoding.countPositives(bytes, offset, length);
             }
-            public String newStringNoRepl(byte[] bytes, Charset cs) throws CharacterCodingException  {
-                return String.newStringNoRepl(bytes, cs);
+            public String newStringReportError(byte[] bytes, Charset cs) throws CharacterCodingException  {
```
Comment (turbanoff) [inline src/java.base/share/classes/java/lang/System.java:2476]:
```suggestion
            public String newStringReportError(byte[] bytes, Charset cs) throws CharacterCodingException {
```

**Pull request closed without merging**